### PR TITLE
Merge branch dev into develop

### PR DESCRIPTION
**Pull request by Idris0v**
## Summary by CodeRabbit

- New Features
  - Background daemon that syncs latest VM releases to local storage with checksum validation.
  - gRPC endpoint to request a specific release and return its local path.
  - Graceful shutdown handling and optional one-time execution mode.

- Build
  - Debian package creation with systemd service, lifecycle scripts, and startup script.
  - Versioned builds integrated into CI.

- Chores
  - Added CI workflows for build and lint on pushes, tags, and PRs.
  - Added dependencies and linting tools; formatting/lint make targets.
  - Updated ignore rules for build artifacts and generated files.

### CODE REVIEW

**Comment by Idris0v** [comment]
@coderabbitai review

**Comment by drunkbatya** [comment]
@Idris0v all fixed

**Comment by Idris0v** [comment]
It's a public repo. Please add Readme with description of what this project is about, local development, deployment guides, usage examples etc. (you can ask ai chat)

also wouldnt hurt to add tests to cover various edge and fail cases (GH api, StorJ)

**Comment by Idris0v** [comment]
should license be GPL-3.0? we use BSL license in all of our open source code

File: app/main.py [hunk 1]
```diff
@@ -2,33 +2,79 @@
...
-    logging.basicConfig(level=level, format="%(asctime)s %(levelname)s %(message)s")
+    logging.basicConfig(level=level, format="%(asctime)s [%(name)s] [%(levelname)s]: %(message)s")
+
 
 def main():
     args = parseArgs()
     init_logging(args.log_level)
+
+    gh = github.Github()
+    ls = local_storage.LocalStorage(basedir="/var/lib/sp/images")
```
Comment (Villain88) [inline app/main.py:41]: I think that hardcoding the path to the download directory is a bad idea, considering that /var may be part of the / mount point and may not always have enough space.

File: app/main.py [hunk 2]
```diff
@@ -2,33 +2,79 @@
...
+    sj = storj.StorJ(
+        token="1UXqNMwov41q9TgHmyopNg5q2giQ8aTdh1gjKWKjfbWPFrcrnhenp6QZfd5ukyVnYXDx9Cok6RtnQMMnXmoZPrSUMNGZGF9KuLCzvRNmQYHowX14C2xAxtJeH6VCuNX39ist4bRE9L5VT3k41frDVh3cG1gZvsqh4EaDeaJyV6U4xVaqXqULnSb9PozqU97VVLWhfwdnj6XgUM59Wzq7yo7vn8RxwSyn8H74TEiLNGUPPA3frsYZuoqWQkNzbiYev5ByWeLro1TXo7DogD4WALCKfEmpwHs9j9rsX5WZvvZ13ourTiuZp5vTTZkByB2ibxUJqkSoZSpCNVtmDToNVKkMREVySe"
```
Comment (Villain88) [inline app/main.py:43]: I agree with coderabit, it might make sense to move the token to the configuration

File: app/modules/github.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
+import logging
```
Comment (Villain88) [inline app/modules/github.py:1]: Just a suggestion:
Use the PyGithub library instead of raw requests.

File: app/main.py [hunk 4]
```diff
@@ -2,33 +2,79 @@
...
-from modules import utils
 
 def parseArgs() -> argparse.Namespace:
     parser = argparse.ArgumentParser(
         description="Daemon to automatically download SuperProtocol VM images",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
     )
     parser.add_argument("--update-time", help="Update check date and time", default="00:00")
-    parser.add_argument("--keep-versions", help="How many previous versions should be keeped", type=int, default=3)
+    parser.add_argument(
```
Comment (Villain88) [inline app/main.py:18]: for future use?

**Comment by drunkbatya** [comment]
> should license be GPL-3.0? we use BSL license in all of our open source code

Replaced

File: app/main.py (hunk 4 above)
Comment (drunkbatya) [reply app/main.py:18]: Yep
